mcs/build/rules.make

```diff
@@ -48,7 +48,7 @@ CSCOMPILE = $(Q_MCS) $(MCS) $(USE_MCS_FLAGS)
 CSC_RUNTIME_FLAGS = --aot-path=$(abspath $(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)) --gc-params=nursery-size=64m
 BASCOMPILE = $(MBAS) $(USE_MBAS_FLAGS)
 CCOMPILE = $(CC) $(USE_CFLAGS)
-BOOT_COMPILE = $(Q_MCS) $(BOOTSTRAP_MCS) $(USE_MCS_FLAGS)
+BOOT_COMPILE = MONO_DEBUG=clr-memory-model $(Q_MCS) $(BOOTSTRAP_MCS) $(USE_MCS_FLAGS)
 INSTALL = $(SHELL) $(topdir)/../mono/install-sh
```

Comment:
Shouldn't this be only enabled on arm ?

Reply:
Yeah, but I didn't bother. On x86/amd64 this option shouldn't lead to any code generation change since the stores on those arches have release semnatics, and OP_MEMORY_BARRIER is ignored. I'm fine either way

Reply:
You also need to bump monolite minimum version when bumping bootstrapping requirements